mp/src/game/client/neo/ui/neo_hud_ammo.cpp

```diff
@@ -137,169 +123,165 @@ void CNEOHud_Ammo::DrawAmmo() const
+	const int maxSpaceAvaliableForBullets = digit_max_width;
+	const int bulletWidth = surface()->GetCharacterWidth(m_hBulletFont, *ammoChar);
+	const int plusWidth = surface()->GetCharacterWidth(m_hBulletFont, '+');
+	const int maxBulletsWeCanDisplay = (maxSpaceAvaliableForBullets / bulletWidth);
+	const int maxBulletsWeCanDisplayWithPlus = ((maxSpaceAvaliableForBullets - plusWidth) / bulletWidth);
+	const int bulletsOverflowing = maxBulletsWeCanDisplay < magSizeMax;
 
-		if(bulletsOverflowing)
-		{
-			magSizeMax = maxBulletsWeCanDisplayWithPlus + 1;
-		}
+	if(bulletsOverflowing)
+	{
+		magSizeMax = maxBulletsWeCanDisplayWithPlus + 1;
+	}
 
-		magSizeMax = min(magSizeMax, 64);
-		char bullets[64]{ '\0' };
-		for(int i = 0; i < magSizeMax; i++)
-		{
-			bullets[i] = *ammoChar;
-		}
+	magSizeMax = min(magSizeMax, 100);
+	char bullets[100]{ '\0' };
+	for(int i = 0; i < magSizeMax; i++)
+	{
+		bullets[i] = *ammoChar;
+	}
 
-		int magAmountToDrawFilled = magSizeCurrent;
+	int magAmountToDrawFilled = magSizeCurrent;
 		
-		if(bulletsOverflowing)
+	if(bulletsOverflowing)
+	{
+		bullets[magSizeMax - 1] = '+';
+
+		if(maxClip == magSizeCurrent)
 		{
-			bullets[magSizeMax - 1] = '+';
-
-			if(maxClip == magSizeCurrent)
-			{
-				magAmountToDrawFilled = magSizeMax;
-			} else if(magSizeMax - 1 < magSizeCurrent)
-			{
-				magAmountToDrawFilled = magSizeMax - 1;
-			} else
-			{
-				magAmountToDrawFilled = magSizeCurrent;
-			}
+			magAmountToDrawFilled = magSizeMax;
+		} else if(magSizeMax - 1 < magSizeCurrent)
+		{
+			magAmountToDrawFilled = magSizeMax - 1;
+		} else
+		{
+			magAmountToDrawFilled = magSizeCurrent;
 		}
+	}
 		
-		wchar_t unicodeBullets[64];
-		g_pVGuiLocalize->ConvertANSIToUnicode(bullets, unicodeBullets, sizeof(unicodeBullets));
+	wchar_t unicodeBullets[100];
+	g_pVGuiLocalize->ConvertANSIToUnicode(bullets, unicodeBullets, sizeof(unicodeBullets));
 		
-		if (magAmountToDrawFilled > 0)
-		{
-			surface()->DrawSetTextFont(m_hBulletFont);
-			surface()->DrawSetTextPos(digit_xpos + xpos, digit_ypos + ypos);
-			surface()->DrawPrintText(unicodeBullets, magAmountToDrawFilled);
-		}
+	if (magAmountToDrawFilled > 0)
+	{
+		surface()->DrawSetTextFont(m_hBulletFont);
+		surface()->DrawSetTextPos(digit_xpos + xpos, digit_ypos + ypos);
+		surface()->DrawPrintText(unicodeBullets, magAmountToDrawFilled);
+	}
 
-		if(maxClip > 0)
-		{
-			if (magSizeMax > 0) {
-				surface()->DrawSetTextColor(textColorTransparent);
-				surface()->DrawSetTextPos(digit_xpos + xpos, digit_ypos + ypos);
-				surface()->DrawPrintText(unicodeBullets, magSizeMax);
-			}
+	if(maxClip > 0)
+	{
+		if (magSizeMax > 0) {
+			const wchar_t* firstEmptyRound = unicodeBullets + magAmountToDrawFilled;
+			surface()->DrawSetTextColor(transparent_ammo_color);
+			surface()->DrawSetTextPos(digit_xpos + xpos + (bulletWidth * magAmountToDrawFilled), digit_ypos + ypos);
+			surface()->DrawPrintText(firstEmptyRound, magSizeMax - magAmountToDrawFilled);
```

Comment:
Sorry to the reviewers for so many line changes, I'm not a fan of deep nesting. This bit is what actually fixes the bullets left being so bright